pandas/_libs/tslibs/offsets.pyx

```diff
@@ -458,7 +458,17 @@ cdef class BaseOffset:
             FutureWarning,
             stacklevel=1,
         )
-        return self.apply(other)
+        return self._apply(other)
+
+    def apply(self, other):
+        # GH#44522
+        warnings.warn(
+            f"{type(self).__name__}.apply is deprecated and will be removed "
+            "in a future version. Use `offset + other` instead",
+            FutureWarning,
+            stacklevel=2,
```

Comment:
use the find_stack_level

Reply:
IIRC find_stack_level doesn't play nicely with cython methods

Reply:
oh interesting, ok